falcon/api_client.go

```diff
@@ -30,11 +32,22 @@ func NewClient(ac *ApiConfig) (*client.CrowdStrikeAPISpecification, error) {
 
 type roundTripper struct{
 	T http.RoundTripper
+	LastRateLimitDigits int
 }
 
 func (rt *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	req.Header.Add("User-Agent", userAgent)
-	return rt.T.RoundTrip(req)
+
+	if rt.LastRateLimitDigits == 1 || rt.LastRateLimitDigits == 2 {
+		log.Debug("Chances are relatively high of next API call to be rejected by X-Ratelimit. Waiting 500 milisecond.")
+		time.Sleep(500 * time.Millisecond)
```

Comment:
```suggestion
	if rt.LastRateLimitDigits == 1 || rt.LastRateLimitDigits == 2 {
		log.Debug("Approaching API rate limits. Waiting 500 milisecond.")
		time.Sleep(500 * time.Millisecond)
```

Reply:
Suggesting language to make it more explicit the rate limits are being approached